geniza/corpus/forms.py

```diff
@@ -0,0 +1,75 @@
+from django import forms
+
+
+class SelectDisabledMixin:
+    """
+    Mixin for :class:`django.forms.RadioSelect` or :class:`django.forms.CheckboxSelect`
+    classes to set an option as disabled. To disable, the widget's choice
+    label option should be passed in as a dictionary with `disabled` set
+    to True::
+
+        {'label': 'option', 'disabled': True}.
+    """
+
+    # copied from mep-django
+
+    # Using a solution at https://djangosnippets.org/snippets/2453/
+    def create_option(
+        self, name, value, label, selected, index, subindex=None, attrs=None
+    ):
+        disabled = None
+
+        if isinstance(label, dict):
+            label, disabled = label["label"], label.get("disabled", False)
+        option_dict = super().create_option(
+            name, value, label, selected, index, subindex=subindex, attrs=attrs
+        )
+        if disabled:
+            option_dict["attrs"].update({"disabled": "disabled"})
+        return option_dict
+
+
+class SelectWithDisabled(SelectDisabledMixin, forms.Select):
+    """
+    Subclass of :class:`django.forms.Select` with option to mark
+    a choice as disabled.
+    """
+
+
+class DocumentSearchForm(forms.Form):
+    query = forms.CharField(
+        label="Keyword or Phrase",
+        required=False,
+        widget=forms.TextInput(
+            attrs={
+                "placeholder": "search by keyword",
+                "aria-label": "Keyword or Phrase",
+            }
+        ),
```

Comment:
I tested adding `"type": "search",` here to get [search input behavior](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search), which is nice – it remembers previous entries, and on some platforms also adds a little search indicator or "clear field" button on its own.